addons/sale_timesheet/models/sale_order.py

```diff
@@ -280,5 +280,12 @@ def _recompute_qty_to_invoice(self, start_date, end_date):
             domain = expression.AND([domain, [('date', '<=', end_date)]])
         mapping = lines_by_timesheet.sudo()._get_delivered_quantity_by_analytic(domain)
 
+        # we need to remove the already invoiced quantity for lines that were credited
+        if refund_account_moves:
+            for sol_id in mapping:
+                line = refund_account_moves.line_ids.sale_line_ids.filtered(lambda sol: sol.id == sol_id)
+                if line:
+                    mapping[sol_id] -= line.qty_invoiced
```

Comment:
And if we already invoiced something, we will remove more than expected, won't we? 🤔

Reply:
At first in the mapping the value associated with the line is the delivered_quantity, by removing the quantity that was already invoiced from the delivered_quantity, we get the difference, which is the quantity we want to invoice.
If a line is already fully invoiced, the difference between the delivered_quantity and invoiced_quantity is 0, and when at the line `line.qty_to_invoice = ...`(L:291) is set to 0, the line is not included in the generated invoice.